polaris/evaluate/_predictions.py

```diff
@@ -0,0 +1,197 @@
+                    "should be a dictionary with the test set labels as keys."
+                )
+            predictions = {k: {target_labels[0]: v} for k, v in predictions.items()}
+
+        # (4) Multi-task, single test set: We expect a dictionary with the target labels as keys.
+        elif not is_single_task and is_single_test:
+            if not isinstance(predictions, dict) or set(predictions.keys()) != set(target_labels):
+                raise ValueError(
+                    "The predictions for multi-task, single test set benchmarks "
+                    "should be a dictionary with the target labels as keys."
+                )
+            predictions = {test_set_labels[0]: predictions}
+
+        # (5) Multi-task, multi-test sets: The predictions should be fully-specified
+        else:
+            raise ValueError(
+                "The predictions for multi-task, multi-test sets benchmarks should be fully-specified "
+                "as a nested, two-level dictionary: { test_set_name: { target_column: np.ndarray } }"
+            )
+
+        return predictions
+
+    @classmethod
+    def _is_fully_specified(
+        cls, predictions: IncomingPredictionsType, target_labels: list[str], test_set_labels: list[str]
+    ) -> bool:
+        """
+        Check if the predictions are fully specified for the target columns and test set names.
+        """
+        # Not a dictionary
+        if not isinstance(predictions, dict):
+            return False
+
+        # Outer-level of the dictionary should correspond to the test set names
+        if set(predictions.keys()) != set(test_set_labels):
+            return False
+
+        # Inner-level of the dictionary should correspond to the target columns
+        for _, test_set_predictions in predictions.items():
+            if not isinstance(test_set_predictions, dict):
+                return False
+            if set(test_set_predictions.keys()) != set(target_labels):
+                return False
+
+        return True
+
+    @classmethod
+    def _check_test_set_size(
+        cls, predictions: PredictionsType, test_set_sizes: dict[str, int]
+    ) -> PredictionsType:
+        for test_set_label, test_set in predictions.items():
+            for target in test_set.values():
+                if test_set_label not in test_set_sizes:
+                    raise ValueError(f"Expected size for test set '{test_set_label}' is not defined")
+
+                if len(target) != test_set_sizes[test_set_label]:
+                    raise ValueError(
+                        f"Predictions size mismatch: The predictions for test set '{test_set_label}' "
+                        f"should have a size of {test_set_sizes[test_set_label]}, but have a size of {len(target)}."
+                    )
+        return predictions
+
+    @classmethod
+    def _convert_lists_to_arrays(cls, predictions: IncomingPredictionsType) -> IncomingPredictionsType:
```

Comment:
**Suggestion**
This could go outside the class. Maybe even in the `utils`module.